lib/requester.js

```diff
@@ -22,57 +22,63 @@ module.exports = function(requestObject, debug, cb){
   method = requestObject.method && requestObject.method.toLowerCase(),
   headers = headersToRequestFormat(requestObject.headers),
   json, rawHttpRequest, rawHttpResponse, r;
-  
+
   // Ensure we convert out of a mongoose document
   if (requestObject.toObject){
     requestObject = requestObject.toObject();
   }
-  
+
   if (_.isFunction(debug)){
     cb = debug;
     debug = false;
   }
-  
+
   if (!requestObject.url || !method){
     return cb({ code : 400, message : 'All requests must contain a URL and Method' });
   }
-  
+
   if (!request.hasOwnProperty(method)){
     return cb({code : 405, message : 'Method ' + method + ' is not supported'});
   }
-  
-  if (requestObject.body && _.isObject(requestObject.body)){
+
+  if(_.isObject(requestObject.body))
+  {
     requestObject.body = JSON.stringify(requestObject.body);
   }
-  
+
   if (headers.hasOwnProperty('content-type') && headers['content-type'] === 'application/json'){
```

Comment:
A bit difficult to explain but, this issue was with a double stringify occurring. Once by our code, and another through a node module, which is not clear to see by looking at the code fragment above.

The invalid JSON error is thrown by the inputted JSON beginning with a symbol other than '{'. 
 
Here, if the body of the request is `{ "barcode" : "077924004971" }`, this code below will convert it to `{"barcode":"077924004971"}`
```
if(_.isObject(requestObject.body))
  {
    requestObject.body = JSON.stringify(requestObject.body);
 }
```

The next block of code checks if the body contains JSON, and if it does, If the body is of type string (ie. its already been double stringified), so, to prevent the error been thrown where the first character begins with a `"` and not a `{`, we parse it. This block would convert `"{ \"barcode\" : \"077924004971\" }"` to ``{ "barcode" : "077924004971" }``. This will then satisfy the valid JSON checks in the body-parse module.
```
if (headers.hasOwnProperty('content-type') && headers['content-type'] === 'application/json'){
    json = true;

    if(typeof requestObject.body === "string")
    {
      requestObject.body = JSON.parse(requestObject.body);
    }
  }
```

All in all, the JSON.parse is only being used on doubely stringified (?) request bodies. 

Hopefully I explained that ok.